Tribler/Core/Modules/MetadataStore/OrmBindings/channel_metadata.py

```diff
@@ -0,0 +1,259 @@
+    # optimize_alignment + merkle + mutable_torrent_support = 19
+    # t = create_torrent(fs, flags=19) # BUG?
+    t.set_priv(False)
+    set_piece_hashes(t, os.path.dirname(directory))
+    torrent = t.generate()
+    with open(torrent_filename, 'wb') as f:
+        f.write(bencode(torrent))
+
+    infohash = torrent_info(torrent).info_hash().to_bytes()
+    return torrent, infohash
+
+
+def define_binding(db):
+    class ChannelMetadata(db.TorrentMetadata):
+        _discriminator_ = MetadataTypes.CHANNEL_TORRENT.value
+        version = orm.Optional(int, size=64, default=0)
+        subscribed = orm.Optional(bool, default=False)
+        votes = orm.Optional(int, size=64, default=0)
+        _payload_class = ChannelMetadataPayload
+        _channels_dir = None
+
+        @db_session
+        def update_metadata(self, update_dict=None):
+            now = datetime.utcnow()
+            channel_dict = self.to_dict()
+            channel_dict.update(update_dict or {})
+            channel_dict.update({
+                "size": len(self.contents_list),
+                "timestamp": now,
+            })
+            self.set(**channel_dict)
+            self.sign()
+
+        @classmethod
+        @db_session
+        def process_channel_metadata_payload(cls, payload):
+            """
+            Process some channel metadata.
+            :param payload: The channel metadata, in serialized form.
+            :return: The ChannelMetadata object that contains the latest version of the channel
+            """
+            channel = ChannelMetadata.get_channel_with_id(payload.public_key)
+            if channel:
+                if payload.version > channel.version:
+                    channel.delete()
+                    return ChannelMetadata.from_payload(payload)
+                else:
+                    return channel
+            else:
+                return ChannelMetadata.from_payload(payload)
+
+        @classmethod
+        @db_session
+        def create_channel(cls, title, description):
+            """
+            Create a channel and sign it with a given key.
+            :param title: The title of the channel
+            :param description: The description of the channel
+            :return: The channel metadata
+            """
+            if ChannelMetadata.get_channel_with_id(cls._my_key.pub().key_to_bin()):
+                raise DuplicateChannelNameError()
+
+            my_channel = cls(public_key=buffer(cls._my_key.pub().key_to_bin()), title=title,
+                             tags=description, subscribed=True)
+            my_channel.sign()
+            return my_channel
+
+        def consolidate_channel_torrent(self):
+            """
+            Delete the channel dir contents and create it anew.
+            Use it to consolidate fragmented channel torrent directories.
+            :param key: The public/private key, used to sign the data
+            """
+
```

Comment:
Minor comment: you can remove this enter.

Reply:
I always go through pylint after the PR is accepted, so don't worry about these minor stuff ))